Lib/unittest/test/test_async_case.py

```diff
@@ -0,0 +1,151 @@
+import asyncio
+import unittest
+
+
+def tearDownModule():
+    asyncio.set_event_loop_policy(None)
+
+
+class TestAsyncCase(unittest.TestCase):
+    def test_basic(self):
+        events = []
+
+        class Test(unittest.AsyncioTestCase):
+            async def setUp(self):
+                self.assertEqual(events, [])
+                events.append('setUp')
+
+            async def test_func(self):
+                self.assertEqual(events, ['setUp'])
+                events.append('test')
+                self.addCleanup(self.on_cleanup)
+
+            async def tearDown(self):
+                self.assertEqual(events, ['setUp', 'test'])
```

Comment:
One of the problems I can see happening here is multiple inheritance of tests cases with both "async" and "sync" versions of "tearDown". 
* How are you supposed to call `super().tearDown()` in such case?  
* Is this a serious problem?  I think it might be, essentially we're breaking the cooperative nature of `super()`.
* Should we prefix all "async" hooks, e.g. "async def asyncTearDown"? Or is that too ugly?